tensorflow/compiler/xla/service/gpu/multi_output_fusion.cc

```diff
@@ -47,13 +47,8 @@ bool GpuMultiOutputFusion::ShapesCompatibleForFusion(HloInstruction* instr1,
 bool GpuMultiOutputFusion::IsFusible(HloInstruction* instr) {
   // We can fuse reduces and loop fusions. Elementwise instructions can be fused
   // with any other instruction.
-  // TODO(b/112957171): This should use the same isFusible logic as
-  // instruction_fusion.
   return instr->IsFusible() &&
-         (IsInputFusibleReduction(*instr) ||
-          (instr->opcode() == HloOpcode::kFusion &&
-           instr->fusion_kind() == HloInstruction::FusionKind::kLoop) ||
-          instr->IsElementwise());
+         (IsInputFusibleReduction(*instr) || instr->IsLoopFusion());
```

Comment:
This change makes the fusion criteria more restrictive: Unfused element-wise ops are no longer considered fusible.
I probably confused you with my earlier comment. My idea was to call IsLoopFusible instead, which does allow unfused, element-wise ops. However, thinking about this again, it also allows certain non-elmentwise ops. While multi-output fusion should handle them just fine, it's probably a good idea to keep this PR a pure refactoring without any functional changes.
In summary, please keep the 
`|| instr->IsElementwise()`